System.Containers/Image.cs

```diff
@@ -67,15 +69,15 @@ public void SetEntrypoint(string executable, string[]? args = null)
             throw new NotImplementedException("Expected base image to have a config node");
         }
 
-        configObject["Entrypoint"] = executable;
+        configObject["Entrypoint"] = ToJsonArray(executableArgs);
 
         if (args is null)
         {
             configObject.Remove("Cmd");
         }
         else
         {
-            configObject["Cmd"] = new JsonArray(args.Where(s => !string.IsNullOrEmpty(s)).Select(s =>(JsonObject)s).ToArray());
```

Comment:
there was a small bug here - string can't be cast to JsonObject directly, you need to cast to JsonValue.